### PR TITLE
chore: contributing section in readme.md

### PR DESCRIPTION
**Pull request by Meng-Heng**
Fixes: #2365

This is how much I came up with the synopsis so far

### CODE REVIEW

**Comment by keyman-server** [comment]
Thank you for your pull request. The Keyman keyboard review team have been notified of your pull request and will review it and build it shortly.

**Comment by LornaSIL** [comment]
I wonder if it would be helpful to add something like: "Make sure your keyboard, and the package, compile before you make your submission. If there are warning messages, try to address all of them; if there are errors those should be corrected before submission."

File: README.md [hunk 1]
```diff
@@ -18,9 +18,23 @@ for example `sil` or `gff`.
...
+
+A good way to start creating a keyboard is to plan out the requirements of the keyboard ahead and decide on the keyboard's language, target users, usability, structure, and many more steps that can be carried out further along the process. To ensure that the keyboard is up to standard, start by implementing the D.I.S.C.U.S. method. Read more! <https://help.keyman.com/developer/keyboards/standard/>
+
+When a keyboard is ready to publish, please confirm that the folders and files are consistent to the repository. For more clarification: <https://help.keyman.com/developer/keyboards/github/step-0>
+
+Learn how to update existing keyboards <https://community.software.sil.org/t/unfamiliar-with-git-best-way-to-update-my-keyboard/8652/2>
+
+### Keyboard Help Documentation
+
+If the guides are not enough to provide you a completed keyboard, don't be afriad to explore the main keyboard help page! <https://help.keyman.com/developer/keyboards/>
```
Comment (Meng-Heng) [inline README.md:37]:
```suggestion
If the guides are not enough to provide you a completed keyboard, don't be afraid to explore the main keyboard help page! <https://help.keyman.com/developer/keyboards/>
```

File: README.md [hunk 2]
```diff
@@ -18,9 +18,23 @@ for example `sil` or `gff`.
...
 
-Keyman Developer 17.0+ is recommended to edit these files. <https://keyman.com/developer>
+### Tool
+
+Get the latest version of Keyman Developer to start the journey of keyboard development: <https://keyman.com/developer>
+
+### Keyboard Design Principles
+
+A good way to start creating a keyboard is to plan out the requirements of the keyboard ahead and decide on the keyboard's language, target users, usability, structure, and many more steps that can be carried out further along the process. To ensure that the keyboard is up to standard, start by implementing the D.I.S.C.U.S. method. Read more! <https://help.keyman.com/developer/keyboards/standard/>
+
```
Comment (Meng-Heng) [inline README.md:30]:
```suggestion
Please make sure the keyboard and the package compile successfully before making a keyboard submission. If there are warning messages, try to address all of them; if there are errors, those must be corrected before submission. Help on this: <https://help.keyman.com/developer/17.0/reference/messages/>
```

Comment (Meng-Heng) [reply README.md:30]: Thanks @LornaSIL. This is perfect between the keyboard design and keyboard submission.

**Comment by keyman-server** [comment]
Thank you for your pull request. The Keyman keyboard review team have been notified of your pull request and will review it and build it shortly.

File: README.md [hunk 3]
```diff
@@ -18,9 +18,23 @@ for example `sil` or `gff`.
...
+When a keyboard is ready to publish, please confirm that the folders and files are consistent to the repository. For more clarification: <https://help.keyman.com/developer/keyboards/github/step-0>
```
Comment (Meng-Heng) [inline README.md:31]:
```suggestion
When a keyboard is ready to publish, please confirm that the folders and files are consistent to the repository. For more clarification: 
- <https://help.keyman.com/developer/current-version/reference/file-layout>
- <https://help.keyman.com/developer/current-version/reference/file-types/>
- <https://help.keyman.com/developer/keyboards/github/step-0>

```

File: README.md [hunk 4]
```diff
@@ -18,9 +18,23 @@ for example `sil` or `gff`.
...
+
+Learn how to update existing keyboards <https://community.software.sil.org/t/unfamiliar-with-git-best-way-to-update-my-keyboard/8652/2>
+
+### Keyboard Help Documentation
+
+If the guides are not enough to provide you a completed keyboard, don't be afraid to explore the main keyboard help page! <https://help.keyman.com/developer/keyboards/>
```
Comment (Meng-Heng) [inline README.md:37]:
```suggestion
If the guides are not enough to provide you with a completed keyboard, don't be afraid to explore the main keyboard and keyman developer help pages! 
- <https://help.keyman.com/developer/keyboards/>
- <https://help.keyman.com/developer/current-version/reference/>

```

File: README.md [hunk 5]
```diff
@@ -18,9 +18,23 @@ for example `sil` or `gff`.
 
 ## Contributing
 
-First, read the guide! <https://help.keyman.com/developer/keyboards/>
+Standard keyboard follows a structure of user-friendly design, keyboard structures, and typing experience.
```
Comment (Meng-Heng) [inline README.md:21]:
```suggestion

```

File: README.md [hunk 6]
```diff
@@ -104,4 +118,6 @@ new keyboard name.
 * <https://help.keyman.com/keyboards/>: Published documentation for each keyboard
 * <https://r.keymanweb.com/api/4.0/keyboards/>: JSON API for compiled keyboards available from Keyman Cloud
 
+## Community
 
+We appreciate the effort of reaching out about any issue or suggestion to help us improve the experience and keyboards. Share with us on our community site: <https://community.software.sil.org/new-message?groupname=keyman>
```
Comment (Meng-Heng) [inline README.md:123]:
```suggestion
We appreciate the effort of reaching out about any issue or suggestion that is beneficial for the community. Share with us on our community site: <https://community.software.sil.org/new-message?groupname=keyman>
```

**Comment by keyman-server** [comment]
Thank you for your pull request. The Keyman keyboard review team have been notified of your pull request and will review it and build it shortly.

File: README.md [hunk 7]
```diff
@@ -18,9 +18,29 @@ for example `sil` or `gff`.
...
 
-Keyman Developer 17.0+ is recommended to edit these files. <https://keyman.com/developer>
+### Tool
+
+Get the latest version of Keyman Developer to start the journey of keyboard development: <https://keyman.com/developer>
+
+### Keyboard Design Principles
+
+A good way to start creating a keyboard is to plan out the requirements of the keyboard ahead and decide on the keyboard's language, target users, usability, structure, and many more steps that can be carried out further along the process. To ensure that the keyboard is up to standard, start by implementing the D.I.S.C.U.S. method. Read more! <https://help.keyman.com/developer/keyboards/standard/>
+Please make sure the keyboard and the package compile successfully before making a keyboard submission. If there are warning messages, try to address all of them; if there are errors, those must be corrected before submission. Help on this: <https://help.keyman.com/developer/current-version/reference/messages/>
```
Comment (Meng-Heng) [inline README.md:29]:
```suggestion

Please make sure the keyboard and the package compile successfully before making a keyboard submission. If there are warning messages, try to address all of them; if there are errors, those must be corrected before submission. Help on this: <https://help.keyman.com/developer/current-version/reference/messages/>

```

**Comment by keyman-server** [comment]
Thank you for your pull request. The Keyman keyboard review team have been notified of your pull request and will review it and build it shortly.

**Review by LornaSIL — COMMENTED** [review]
You've got this which is very useful:

>Learn how to update existing keyboards <https://community.software.sil.org/t/unfamiliar-with-git-best-way-to-update-my-keyboard/8652/2>

but, in fact, the steps are updated throughout that thread based on the feedback the user gave. It might be best to add that text (and all the updated conversation) into this document.

File: README.md [hunk 8]
```diff
@@ -1,5 +1,22 @@
 # Open Source Keyman keyboards
 
+## Table of Contents
+- [File Layout](#file-layout)
+- [Contributing](#contributing)
+  - [Tools](#tools)
```
Comment (Meng-Heng) [inline README.md:6]:
```suggestion
  - [Tool](#tool)
```

File: README.md [hunk 9]
```diff
@@ -1,5 +1,22 @@
...
+  - [Standard keyboard design](#standard-keyboard-design)
+  - [Keyboard submission guidelines](#keyboard-submission-guidelines)
+  - [Publish your keyboard](#publish-your-keyboard)
+  - [Modify a keyboard layout](#modify-a-keyboard)
+  - [Documentation](#documentation)
+- [Building Keyboards](#building-keyboards)
+  - [Prerequisites](#prerequisites)
+  - [Build instructions](#build-instructions)
+  - [Common `build.sh` parameters](#common-buildsh-parameters)
+- [Deprecated Keyboards](#license)
```
Comment (Meng-Heng) [inline README.md:16]:
```suggestion
- [Deprecated Keyboards](#deprecated-keyboards)
```

**Comment by keyman-server** [comment]
Thank you for your pull request. The Keyman keyboard review team have been notified of your pull request and will review it and build it shortly.

**Comment by keyman-server** [comment]
Thank you for your pull request. The Keyman keyboard review team have been notified of your pull request and will review it and build it shortly.

**Comment by LornaSIL** [comment]
Just a few more things:
> The files for the keyboard are zipped up and offers for download as: keymanapp keyboards master release-a_ausephon.zip

Change "offers" to "offered"

> only specifying the exisiting shared-font location in the package is neccesarily.

Change "neccesarily" to "necessary"

Thanks. This is very useful.

File: README.md [hunk 10]
```diff
@@ -18,9 +35,74 @@ for example `sil` or `gff`.
...
+  - Update the forked repository [here](https://help.keyman.com/developer/keyboards/github/step-3#toc-updating-your-fork), and proceed to the next step.
+
+2. Download a copy of the latest files that are in your forked repository as a ZIP file.
+  - Go to `https://github.com/[YourGitHubName]/keyboards/` (after replacing `[YourGitHubName]` with your actual GitHub name)
+  - Navigate to the keyboard, for example, to “release”, then “a”, then “ausephon”
+  - At that point, the browser URL will be something like: `https://github.com/[YourGitHubName]/keyboards/tree/master/release/a/ausephon`
+  - Copy the URL
+  - Go to: <https://download-directory.github.io/>
+  - Paste the URL into the field and press Enter
+  - The files for the keyboard are zipped up and offers for download as: `keymanapp keyboards master release-a_ausephon.zip`
```
Comment (Meng-Heng) [inline README.md:76]:
```suggestion
  - The files for the keyboard are zipped up and offered for download as: `keymanapp keyboards master release-a_ausephon.zip`
```

File: README.md [hunk 11]
```diff
@@ -18,9 +35,74 @@ for example `sil` or `gff`.
...
+  - Go back to GitHub
+  - Create a new branch (from master) with a short name to represent the pull request. For example, ausephon-updates...
+  - Navigate to the target folder location (click on release, then on a, and you’ll see the ausephon folder)
+  - Click on “Add file”, then “Upload files”
+  - Drag the “ausephon” folder from your Windows Explorer window, and drop it on the GitHub window
+  - Follow the steps to create a pull request, although the commit message will be something like `[ausephon] updated keyboard`
+
+### Include font files
+
+If the fonts are used in multiple keyboards, the font files must be placed under the `shared folder`. Follow through this [documentation](https://help.keyman.com/developer/keyboards/github/step-2#toc-including-fonts-in-your-keyboard-package). However, if you are editing the keyboard package within the folders of the forked repository (skip step 2 above) only specifying the exisiting shared-font location in the package is neccesarily.
```
Comment (Meng-Heng) [inline README.md:98]:
```suggestion
If the fonts are used on multiple keyboards, the font files must be placed under the `shared folder`. Follow through this [documentation](https://help.keyman.com/developer/keyboards/github/step-2#toc-including-fonts-in-your-keyboard-package). However, if you are editing the keyboard package within the folders of the forked repository (skip step 2 above) only specifying the existing shared-font location in the package is necessary.
```

**Comment by Meng-Heng** [comment]
Thanks, Lorna! It should be all better now.

**Comment by keyman-server** [comment]
Thank you for your pull request. The Keyman keyboard review team have been notified of your pull request and will review it and build it shortly.

**Review by LornaSIL — APPROVED** [review]
Looks great!

**Pull request merged**